contracts/mesh-provider/src/contract.rs

```diff
@@ -121,6 +123,14 @@ pub fn execute_receive_claim(
     let mut stake = STAKED
         .may_load(deps.storage, (&owner, &validator))?
         .unwrap_or_default();
+
+    // First calculate rewards with old stake (or set default if first delegation)
+    stake.calc_pending_rewards(
```

Comment:
Okay, we should definitely not error if no block has passed (comment in state.rs)
Staking twice a few blocks later is not bad. Just do a no-op if nothing to claim